src/portable/synopsys/dwc2/dcd_dwc2.c

```diff
@@ -878,6 +919,14 @@ void dcd_edpt_close (uint8_t rhport, uint8_t ep_addr)
 
   dcd_edpt_disable(rhport, ep_addr, false);
 
+  #if TU_CHECK_MCU(OPT_MCU_ESP32S2, OPT_MCU_ESP32S3, OPT_MCU_ESP32P4)
```

Comment:
Here, we should: 
1. Release (decrease usage counter) only for endpoints, which are not 0
2. Not decrease the value, if it is already 0 (not to overflow the value)

So, it is better to make another small function (to have a small call here and more clear logic inside the specific function) and call it here. 
Something like: 

```suggestion
  #if TU_CHECK_MCU(OPT_MCU_ESP32S2, OPT_MCU_ESP32S3, OPT_MCU_ESP32P4)
  if (epnum) {
    dcd_ep_release(rhport, dir);
  }
  #endif
```